src/nvim/terminal.c

```diff
@@ -1217,12 +1218,14 @@ static bool is_focused(Terminal *term)
 
 #define GET_CONFIG_VALUE(k, o) \
   do { \
-    Error err; \
+    Error err = ERROR_INIT; \
     /* Only called from terminal_open where curbuf->terminal is the */ \
     /* context  */ \
     o = dict_get_value(curbuf->b_vars, cstr_as_string(k), &err); \
+    api_free_error(err); \
     if (o.type == kObjectTypeNil) { \
       o = dict_get_value(&globvardict, cstr_as_string(k), &err); \
+      api_free_error(err); \
```

Comment:
@oni-link `api_clear_error(Error *)` sets msg to NULL now, so the double-free should be avoided. Thanks for finding that.